gnc/sub8_perception/nodes/torpedo_board_pose_est.py

```diff
@@ -22,105 +63,100 @@ def __init__(self):
+        # TODO: figure out why a negative is needed above, is there a problem with my frames?
+        homogenizer = np.array([[0, 0, 0, 1]])
+        tf_cam_from_tb_hom = np.vstack((
+            np.hstack((
+                rot_tb_from_cam.T,
+                -np.dot(rot_tb_from_cam.T, trans_tb_from_cam))),
+            homogenizer))
+        corners_from_cam_hom = np.dot(tf_cam_from_tb_hom, self.corners_from_tb_hom)
+        return corners_from_cam_hom
 
     def visualize_pose_est(self):
-        pts = self.generate_hom_tb_corners_from_cam(self.prelim_pose_est)
-        p1 = Point(x=pts[0, 0], y=pts[1, 0], z=pts[2, 0])
-        p2 = Point(x=pts[0, 1], y=pts[1, 1], z=pts[2, 1])
-        p3 = Point(x=pts[0, 2], y=pts[1, 2], z=pts[2, 2])
-        p4 = Point(x=pts[0, 3], y=pts[1, 3], z=pts[2, 3])
         marker = visualization_msgs.Marker(
             ns='torpedo_board/pose_est',
             id=0,
-            header=sub8_utils.make_header(frame=self.pose_req_frame),
+            header=sub8_utils.make_header(frame=self.current_req.frame_id),
             type=visualization_msgs.Marker.LINE_STRIP,
             action=visualization_msgs.Marker.ADD,
             color=ColorRGBA(0.0, 1.0, 10, 0.7),
             scale=Vector3(0.05, 0.0, 0.0),
             lifetime=rospy.Duration()
         )
-        marker.points.append(p1)
-        marker.points.append(p2)
-        marker.points.append(p3)
-        marker.points.append(p4)
-        marker.points.append(p1)
+        pts = self.generate_hom_tb_corners_from_cam(self.minimization_result.x)
+        marker.points.append(Point(x=pts[0, 0], y=pts[1, 0], z=pts[2, 0]))
+        marker.points.append(Point(x=pts[0, 1], y=pts[1, 1], z=pts[2, 1]))
+        marker.points.append(Point(x=pts[0, 2], y=pts[1, 2], z=pts[2, 2]))
+        marker.points.append(Point(x=pts[0, 3], y=pts[1, 3], z=pts[2, 3]))
+        marker.points.append(Point(x=pts[0, 0], y=pts[1, 0], z=pts[2, 0]))
         self.marker_pub.publish(marker)
 
 
 class ParsedPoseEstRequest(object):
     def __init__(self, req):
         self.seq = req.pose_stamped.header.seq
-        self.stamp = req.pose_stamped.header.stamp
+        self.stamp = rospy.Time.from_sec(req.pose_stamped.header.stamp.to_time())
         self.frame_id = req.pose_stamped.header.frame_id
         self.position = req.pose_stamped.pose.position
         self.orientation = req.pose_stamped.pose.orientation
-        quat = np.array(
-            [self.orientation.x, self.orientation.y, self.orientation.z, self.orientation.w])
-        self.yaw = tf.transformations.euler_from_quaternion(quat, 'syxz')[0]
+        quat = np.array([self.orientation.x, self.orientation.y, self.orientation.z, self.orientation.w])
+        self.pose = np.array([self.position.x, self.position.y, self.position.z,
+                              tf.transformations.euler_from_quaternion(quat, 'syxz')[0]])
         self.left_cam_matx = np.array([
```

Comment:
If you wanted to, you could change the service to include a camera_info message so that you wouldn't have to rebuild these matricies.

Also, was there a reason for moving the service to here instead of back in sub8_msgs?